commcare_connect/opportunity/tw_form_views.py

```diff
@@ -80,3 +94,166 @@ def verification_flags_config(request, org_slug=None, pk=None):
+    opportunity = get_opportunity_or_404(pk=opp_id, org_slug=org_slug)
+    payment_unit = get_object_or_404(PaymentUnit, id=pk, opportunity=opportunity)
+    deliver_units = DeliverUnit.objects.filter(
+        Q(payment_unit__isnull=True) | Q(payment_unit=payment_unit) | Q(payment_unit__opportunity__active=False),
+        app=opportunity.deliver_app,
+    )
+    exclude_payment_units = [payment_unit.pk]
+    if payment_unit.parent_payment_unit_id:
+        exclude_payment_units.append(payment_unit.parent_payment_unit_id)
+    payment_unit_deliver_units = {deliver_unit.pk for deliver_unit in payment_unit.deliver_units.all()}
+    opportunity_payment_units = (
+        opportunity.paymentunit_set.filter(
+            Q(parent_payment_unit=payment_unit.pk) | Q(parent_payment_unit__isnull=True)
+        )
+        .exclude(pk__in=exclude_payment_units)
+        .all()
+    )
+    form = PaymentUnitForm(
+        deliver_units=deliver_units,
+        instance=payment_unit,
+        data=request.POST or None,
+        payment_units=opportunity_payment_units,
+    )
+    if form.is_valid():
+        form.save()
+        required_deliver_units = form.cleaned_data["required_deliver_units"]
+        DeliverUnit.objects.filter(id__in=required_deliver_units).update(payment_unit=form.instance.id, optional=False)
+        optional_deliver_units = form.cleaned_data["optional_deliver_units"]
+        DeliverUnit.objects.filter(id__in=optional_deliver_units).update(payment_unit=form.instance.id, optional=True)
+        sub_payment_units = form.cleaned_data["payment_units"]
+        PaymentUnit.objects.filter(id__in=sub_payment_units, parent_payment_unit__isnull=True).update(
+            parent_payment_unit=form.instance.id
+        )
+        # Remove deliver units which are not selected anymore
+        deliver_units = required_deliver_units + optional_deliver_units
+        removed_deliver_units = payment_unit_deliver_units - {int(deliver_unit) for deliver_unit in deliver_units}
+        DeliverUnit.objects.filter(id__in=removed_deliver_units).update(payment_unit=None, optional=False)
+        removed_payment_units = {payment_unit.id for payment_unit in opportunity_payment_units} - {
+            int(payment_unit_id) for payment_unit_id in sub_payment_units
+        }
+        PaymentUnit.objects.filter(id__in=removed_payment_units, parent_payment_unit=form.instance.id).update(
+            parent_payment_unit=None
+        )
+        messages.success(request, f"Payment unit {form.instance.name} updated. Please reset the budget")
+        return redirect("opportunity:finalize", org_slug=request.org.slug, pk=opportunity.id)
+    return render(
+        request,
+        "tailwind/pages/form.html",
+        dict(title=f"{request.org.slug} - {opportunity.name}", form_title="Payment Unit Edit", form=form),
+    )
+
+
+@org_admin_required
+def send_message_mobile_users(request, org_slug=None, pk=None):
```

Comment:
I am wondering if we shouldn't copy paste the full view as I did, instead modify existing methods to take a form class arg and use like below

```
@org_admin_required
def send_message_mobile_users(request, org_slug=None, pk=None):
    return _send_message_view(request, org_slug, pk, SendMessageMobileUsersForm)


@org_admin_required
def send_message_mobile_users_tw(request, org_slug=None, pk=None):
    # SendMessageMobileUsersForm from tw_forms
    return _send_message_view(request, org_slug, pk, SendMessageMobileUsersForm)
```

This way any changes done to upstream view, we don't have to worry about them.